kernel/src/sync/mutex.rs

```diff
@@ -75,11 +77,12 @@ impl<T, S: MutexSupport> Mutex<T, S> {
     ///     drop(lock);
     /// }
     /// ```
-    pub fn new(user_data: T) -> Mutex<T, S> {
+    pub const fn new(user_data: T) -> Mutex<T, S> {
         Mutex {
             lock: AtomicBool::new(false),
             data: UnsafeCell::new(user_data),
-            support: S::new(),
+            support: MaybeUninit::uninit(),
```

Comment:
Can we force S::new() to be const fn?

Reply:
Currently it's forbidden to have a const fn within a Trait, so I assume there is no way now.